### DISCUSSION TITLE
Collapse Horizontal

### DISCUSSION THREAD

**Discussion by petermanders89** [mantinedev/mantine#1470] [Feature requests]
Currently out of the box the collapse only supports vertical, the default. It would be more than welcome it also supports horizontal. It kind of works; but all transition/animations are not working correctly. 

An example from [react-bootstrap](https://react-bootstrap.github.io/utilities/transitions/), it supports both. By passing the property  `dimension` with the value `width`, it changes the animations used.

**Comment by iakovalgaev** [comment]
I agree. I also need this opportunity

**Reply by rtivital** [reply]
It is done, will be available soon.

**Comment by fakeplant** [comment]
Any update on when this might be re-added, as it was removed in 5.9.2?

**Reply by rtivital** [reply]
I suppose that it won't be added back – I've given up on this, it does not work correctly.

**Reply by fakeplant** [reply]
All good. Thanks for the update.

**Reply by mateusaires** [reply]
@petermanders89 @fakeplant and @rtivital I don't know if it is still relevant, but I copied thet code from mantine/packages/@mantine/core/src/components/Collapse/Collapse.tsx and made it import a copy of mantine/packages/@mantine/core/src/components/Collapse/use-collapse.ts. In this last file (the use-collapse.ts), i just changed every height/Height string or variable (and also the function signatures) to width/Width and it worked flawlessly...
